### PR TITLE
Bluetooth: Mesh: Large Composition Data Client async API

### PR DESCRIPTION
**Pull request by PavelVPV**
This PR adds asynchronous API to Large Composition Data Client model.

### CODE REVIEW

**Review by Andrewpini — CHANGES_REQUESTED** [review]

**Review by Andrewpini — DISMISSED** [review]

File: subsys/bluetooth/mesh/large_comp_data_cli.c [hunk 1]
```diff
@@ -115,69 +123,35 @@ const struct bt_mesh_model_cb _bt_mesh_large_comp_data_cli_cb = {
 	.init = large_comp_data_cli_init,
 };
 
-static int cli_prepare(void *param, uint32_t op, uint16_t addr)
+int data_get(uint16_t net_idx, uint16_t addr, uint32_t op, uint32_t status_op, uint8_t page,
```
Comment (alxelax) [inline subsys/bluetooth/mesh/large_comp_data_cli.c:126]:
```suggestion
static int data_get(uint16_t net_idx, uint16_t addr, uint32_t op, uint32_t status_op, uint8_t page,
```

**Review by alxelax — CHANGES_REQUESTED** [review]
LGTM just a minor improvement

File: tests/bluetooth/mesh_shell/src/main.c [hunk 2]
```diff
@@ -37,6 +37,10 @@ static struct bt_mesh_sol_pdu_rpl_cli srpl_cli;
 static struct bt_mesh_od_priv_proxy_cli od_priv_proxy_cli;
 #endif
 
+#if defined(CONFIG_BT_MESH_LARGE_COMP_DATA_CLI)
+struct bt_mesh_large_comp_data_cli large_comp_data_cli;
```
Comment (alxelax) [inline tests/bluetooth/mesh_shell/src/main.c:41]: I didn't get why is it required to have two instances of the large composition data client. One here and one here: https://github.com/zephyrproject-rtos/zephyr/pull/56109/files#diff-42905dbde6ea39ed1c83d8b8c3e02994a406fd25479769b71dab84ee2ef0e180R30

Comment (PavelVPV) [reply tests/bluetooth/mesh_shell/src/main.c:41]: Removed one.

**Review by Andrewpini — DISMISSED** [review]

**Comment by Andrewpini** [comment]
I guess the failing twister is non-related. It at least seems so to me.

**Review by Andrewpini — DISMISSED** [review]

**Review by mia-ko — CHANGES_REQUESTED** [review]
Just a few comments :)
Not always reviewing the header files, but once one thing caught my attention, I glanced through the rest as well.

**Review by alxelax — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Pull request merged**